collectors/python.d.plugin/mail_log/mail_log.chart.py

```diff
@@ -0,0 +1,234 @@
+        ]
+    },
+
+    'incoming_statuses': {
+        'options': [None, 'Incoming email statuses', 'messages/s', 'incoming',
+                    'mail_log.incoming_statuses', 'stacked'],
+        'lines': [
+            ['mail_accepted', 'accepted', 'incremental', 1, 1],
+            ['mail_greylisted', 'greylisted', 'incremental', 1, 1],
+            ['mail_temp_rejected', 'temporarily rejected', 'incremental', 1, 1],
+            ['mail_perm_rejected', 'permanently rejected', 'incremental', 1, 1]
+        ]
+    },
+
+    'incoming_codes': {
+        'options': [None, 'Incoming email status codes', 'messages/s', 'incoming',
+                    'mail_log.incoming_codes', 'stacked'],
+        'lines': [
+            ["mail_incoming_codes_2.0.0", '2.0.0', 'incremental', 1, 1]
+        ]
+    },
+
+    'outgoing_statuses': {
+        'options': [None, 'Outgoing email statuses', 'messages/s', 'outgoing',
+                    'mail_log.outgoing_statuses', 'stacked'],
+        'lines': [
+            ['mail_sent', 'sent', 'incremental', 1, 1],
+            ['mail_deferred', 'deferred', 'incremental', 1, 1],
+            ['mail_bounced', 'bounced', 'incremental', 1, 1]
+        ]
+    },
+
+    'outgoing_codes': {
+        'options': [None, 'Outgoing email status codes', 'messages/s', 'outgoing',
+                    'mail_log.outgoing_codes', 'stacked'],
+        'lines': [
+            ["mail_outgoing_codes_2.0.0", '2.0.0', 'incremental', 1, 1]
+        ]
+    },
+}
+
+
+class MailStatistics:
+    # incoming
+    connections = 0
+    incoming_codes = {
+        "2.0.0": 0
+    }
+    accepted = 0
+    temp_rejected = 0
+    perm_rejected = 0
+    greylisted = 0
+    # outgoing
+    outgoing_codes = {
+        "2.0.0": 0
+    }
+    sent = 0
+    deferred = 0
+    bounced = 0
+
+
+class MailParser(object):
+    """Abstract MailParser class, mail log parsers must implement the _parse_line method"""
+
+    def __init__(self, log_service):
+        self.service = log_service
+
+    def _parse_line(self, line):
+        raise NotImplementedError
+
+    def parse_line(self, line):
+        self._parse_line(line)
+
+
+class PostfixParser(MailParser):
+    """MailParser for Postfix"""
+    POSTFIX_RE = re.compile(r'postfix(\/\w+)?\/([a-z]+)\[\d*\]: (\w+)')
+    QUEUE_RE = re.compile(r'from=<.*\(queue active\)')
+    REJECT_RE = re.compile(r'reject: [^:]+: (\d{3} )?(\d+\.\d+\.\d+)')
+    STATUS_RE = re.compile(r'dsn=(\d+\.\d+\.\d+).*status=(\w+)')
+
+    def __init__(self, log_service):
+        super(PostfixParser, self).__init__(log_service)
+        self.greylist_status = log_service.configuration.get("greylist_status", "Try again later")
+        # Since postfix does not log messages passing milters (only rejected ones) we store last 50 ids of messages
+        # received by the `cleanup` process and count them as accepted when queued for delivery for the first time.
+        self.queue_ids = collections.OrderedDict()
```

Comment:
Please dont pass to parser class Service class, if the parser need some config params pass only them.I see it only need `greylist_status`

I think this class should be only resposnible for parsing and populating statistics, i suggest to remove other logic from it. It will help to keep things simple.